mission_control/navigator_missions/nodes/mission_planner.py

```diff
@@ -1,98 +1,176 @@
         self.completeing_mission = False
+        self.base_mission = None
         # TODO Put in YAML file
-        stc = Mission("scan_the_code", ["scan_the_code"], [])
-        bf = Mission("back_and_forth", [], [stc])
-        self.tree.append(bf)
+
+        yaml_text = None
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        yaml_file = dir_path + "/" + yaml_file
+
+        with open(yaml_file, 'r') as stream:
+            try:
+                yaml_text = yaml.load(stream)
+            except yaml.YAMLError as exc:
+                print(exc)
+
+        self.load_missions(yaml_text)
 
     @util.cancellableInlineCallbacks
-    def _init(self):
+    def init_(self):
+        """Initialize the txros aspects of the MissionPlanner b."""
         self.nh = yield NodeHandle.from_argv("mission_planner")
-        global nh
-        nh = self.nh
-        global n
-        n = yield Navigator(nh)._init()
+        self.navigator = yield Navigator(self.nh)._init(True)
+
+        helper = yield DBHelper(self.nh).init_()
+        yield helper.begin_observing(self.new_item)
 
-        self.sub_database = yield nh.subscribe('/database/object_found', PerceptionObject, self.new_item)
-        # self.servcl_exploration_yield = yield nh.get_service_client("/exploration/yield_control", std_msgs.msg.Bool)
+        # This needs to be called in case begin_observing doesn't call refresh
         self.refresh()
         defer.returnValue(self)
 
+    def load_missions(self, yaml_text):
+        """Load all the missions from the YAML file into the structures used in the DAG."""
+        # Load all the missions into a dict mission_name -> Mission
+        # If this mission doesn't have any mission deps, add it to the tree
+        my_missions = {}
+        mission_to_mission_dep = {}
+
+        for name in yaml_text.keys():
+            mission = yaml_text[name]
+            obj_dep = mission["object_dep"]
+            mis_dep = mission["mission_dep"]
+            is_base = mission["is_base"]
+            m = Mission(name, obj_dep)
+            my_missions[name] = m
+            if is_base:
+                self.base_mission = m
+            elif mis_dep == "None":
+                self.tree.append(m)
+            else:
+                mission_to_mission_dep[name] = mis_dep
+
+        # Go through the missions and give them all children dependencies
+        for mission_name in mission_to_mission_dep.keys():
+            parent_name = mission_to_mission_dep[mission_name]
+            child_name = mission_name
+            parent_mission = my_missions[parent_name]
+            child_mission = my_missions[child_name]
+            parent_mission.add_child(child_mission)
+
+    def _is_in_queue(self, x):
+        with self.queue.mutex:
+            return x in self.queue.queue
+
     def new_item(self, obj):
-        self.found.append(obj.name)
+        """Callback for a new object being found."""
+        print "new item, {}".format(obj.name)
```

Comment:
It would be nice if all of these prints were ros logs. This would give us the timestamp and ensure it gets saved to a file. I imagine logging is going to be very important for the mission planner.